src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -61,25 +61,53 @@ public void storeMessage(Message message) {
             .addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
 
-    for (Entity entity : results.asIterable()) {
-      try {
-        String idString = entity.getKey().getName();
-        UUID id = UUID.fromString(idString);
-        String text = (String) entity.getProperty("text");
-        long timestamp = (long) entity.getProperty("timestamp");
-
-        Message message = new Message(id, user, text, timestamp);
-        messages.add(message);
-      } catch (Exception e) {
-        System.err.println("Error reading message.");
-        System.err.println(entity.toString());
-        e.printStackTrace();
-      }
-    }
+    prepareMessages(messages, results);
 
     return messages;
   }
 
+  /**
+   * Gets messages posted by all users.
+   *
+   * @return a list of messages posted by all users, or empty list if no user has ever posted a
+   *     message. List is sorted by time descending.
+   */
+public List<Message> getAllMessages(){
+  List<Message> messages = new ArrayList<>();
+
+  Query query = new Query("Message")
+    .addSort("timestamp", SortDirection.DESCENDING);
+  PreparedQuery results = datastore.prepare(query);
+
+  prepareMessages(messages, results);
```

Comment:
If you change prepareMessages() to return a List<Message> then you can just do "return prepareMessages(results);" here and in getMessages() and remove the messages local variable entirely.

Reply:
Resolved